### ISSUE TITLE
mock db interface

### ISSUE THREAD

**Issue by mariobassem** [threefoldtech/tfgrid-sdk-go#344]
this way we can add unit tests to the http server itself without envolving the database

**Comment by muhamadazmy** [comment]
Yes we discussed this. A good testable code makes user each "unit" can be tested separate from the other components of the system. for example:

An API handler object can own an "abstract" `Storage` object. hence you can test your handler by providing a mock storage that does not actually access a real storage. 

A `Storage` object that is implemented on top of SQL database, can then use an abstract SQL driver hence it does not matter which database is used as long as it is a sql relational database. Hence a test can run against a sqlite database that is filled with fixture data, and so on.
